homeassistant/components/panasonic_viera/__init__.py

```diff
@@ -115,71 +146,106 @@ class Remote:
     """The Remote class. It stores the TV properties and the remote control connection itself."""
 
     def __init__(
-        self, hass, host, port, on_action=None, app_id=None, encryption_key=None,
+        self,
+        hass,
+        host,
+        port,
+        listen_port=DEFAULT_PORT,
+        on_action=None,
+        app_id=None,
+        encryption_key=None,
     ):
         """Initialize the Remote class."""
         self._hass = hass
 
         self._host = host
         self._port = port
 
+        self._listen_port = listen_port
+
         self._on_action = on_action
 
         self._app_id = app_id
         self._encryption_key = encryption_key
 
-        self.state = None
         self.available = False
-        self.volume = 0
-        self.muted = False
-        self.playing = True
+        self.connected = False
+
+        self._state = STATE_OFF
+        self._app_info = None
+        self._volume = "0"
+        self._mute = "0"
+
+        self.playing = False
 
         self._control = None
 
     async def async_create_remote_control(self, during_setup=False):
         """Create remote control."""
-        control_existed = self._control is not None
         try:
             params = {}
             if self._app_id and self._encryption_key:
                 params["app_id"] = self._app_id
                 params["encryption_key"] = self._encryption_key
 
-            self._control = await self._hass.async_add_executor_job(
-                partial(RemoteControl, self._host, self._port, **params)
+            control = await self._hass.async_add_executor_job(
+                partial(
+                    RemoteControl,
+                    self._host,
+                    self._port,
+                    listen_port=self._listen_port,
+                    **params,
+                )
             )
 
-            self.state = STATE_ON
+            await control.async_start_server()
```

Comment:
Why does the library mix an async server with regular I/O?

Reply:
I was in a hurry and couldn't figure out how to do this with a sync library 😂

Since I'll make a new pull request on the library repo, I'll see if I can change this method to an async one

Reply:
It's ok to improve the library later regarding sync vs async.